WordPress/gutenberg
# Adds dimension controls to Group Block

Closes https://github.com/WordPress/gutenberg/issues/14747

![Screen Capture on 2020-02-24 at 15-43-05](https://user-images.githubusercontent.com/444434/75166871-76ad6f80-571c-11ea-998c-fa2333672753.gif)


## Description
Adds dimension controls (currently limited to `padding`) to the Group Block via experimental `DimensionControl` component within the `@wordpress/components` package.


**Please note**: this PR aims to implement the basic feature set for dimension controls. More complex UI will be added at a later date. For a more detailed explanation see [this comment](https://github.com/WordPress/gutenberg/pull/16730#issuecomment-515211211).

## Todo

- [x] Ensure i18n best practices are observed
- [ ] Add deprecation to `core/group` Block
- [x] Add `reset` UI
- [ ] ~Add individual padding side control~ (probably out of scope)
- [ ] ~Enable responsive settings~ (descoped as superceeded by other work in other PRs)

## Screenshots




## How has this been tested?
Manually for now.

To test please use [these test annotations](https://gist.github.com/getdave/a30705607716d08c47046a45a6758c1d).

1. Create a new Page
2. Switch to `Code Editor` mode from the editor settings menu (dots in topmost right corner).
3. Paste in all [the test annotations](https://gist.github.com/getdave/a30705607716d08c47046a45a6758c1d).
4. Switch back to `Visual editor` (again using editor settings menu top right).
5. See all possible variations of dimension combinations.
6. Add backgrounds to make things clearer.

## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
